### PR TITLE
Show a specific icon on external account

### LINKED ISSUE #480: See at glance if an account is external
I'd like to be able to see quickly if accounts are stored externally from the extension (signer or ledger) or if the extension has the keys. I imagine that it could materialize as an icon next to the identicon or the name.

### PR DESCRIPTION
**Pull request by Tbaut**
closes #480 

How it looks like:
external

### CODE REVIEW

File: packages/extension-ui/src/Popup/Forget.tsx [hunk 1]
```diff
@@ -15,6 +15,22 @@ type Props = RouteComponentProps<{ address: string }>;
...
+        res = account.isExternal;
+
+        return true;
+      }
+
+      return false;
+    });
+
+    return res;
+  }, [accounts, address]);
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/Forget.tsx:33]:
```suggestion
  const isExternal = useMemo(
    () => accounts.some((account) =>
      account.address === address && account.isExternal
    )),
    [accounts, address]);
```

I understand the above, mapping to address first, but it seems weird to me. If that approach is really to be followed, either filter + isExternal or find + isExternal.

**Review by jacogr — COMMENTED** [review]
Not sure about the placement? Possibly next to the invisible icon?

**Comment by Tbaut** [comment]
I gave it a shot next to the copy button and the visible indicator.
Placing it here would mean significantly increasing the card size (from 72px to 90+) also the copy and visibility icons are actionable (well the visibility isn't yet, but I think it should, I'll log it) whereas the fact that an account is external is definitely not.
 
image

I took this idea form [fether](https://user-images.githubusercontent.com/6226175/50950496-40079c00-14fe-11e9-9c14-6cb496bf658f.png)

File: packages/extension-ui/src/Popup/Forget.tsx (hunk 1 above)
Comment (Tbaut) [reply packages/extension-ui/src/Popup/Forget.tsx]: thanks, I used find.

**Comment by jacogr** [comment]
Next to the visibility icon? Same line? Agreed that we don't want to make the accounts higher. 

As you suggested elsewhere, we may want to keep that always visible (strike-through vs non) so the external doesn't move around.

**Comment by Tbaut** [comment]
Looks weird to me
image


I'll check it out next to the name

**Comment by Tbaut** [comment]
Here with the name before
image

after
image

**Comment by Tbaut** [comment]
I think my favourite is the arrow before the name. What's your opinion?

**Comment by jacogr** [comment]
Yes, think it works. Would just adjust it to be the same lightness as the visibility one.

File: packages/extension-ui/src/Popup/Forget.tsx [hunk 2]
```diff
@@ -36,7 +36,9 @@ function Forget ({ match: { params: { address } } }: Props): React.ReactElement<
         text={t<string>('Forget account')}
       />
       <div>
-        <Address address={address}>
+        <Address
+          address={address}
+        >
```
Comment (Tbaut) [inline packages/extension-ui/src/Popup/Forget.tsx:41]:
```suggestion
        <Address address={address}>
```

File: packages/extension-ui/src/components/Address.tsx [hunk 3]
```diff
@@ -237,6 +251,7 @@ const Settings = styled.div(({ theme }: ThemeProps) => `
     cursor: pointer;
     background: ${theme.readonlyInputBackground};
   }
+
```
Comment (Tbaut) [inline packages/extension-ui/src/components/Address.tsx:254]:
```suggestion
```

**Comment by Tbaut** [comment]
- Set the color as the label (same as the visibility ones when hovered)
- Added a title attribute, it will surely help some

Here's how it looks now:
external

Selection_266

**Comment by jacogr** [comment]
Ok, let's get it in. I would suggest (once we have ledger) to just make the icons different, QR for the one, something hardware-related for the next.

**Review by jacogr — APPROVED** [review]
lgtm

**Comment by Tbaut** [comment]
> Ok, let's get it in. I would suggest (once we have ledger) to just make the icons different, QR for the one, something hardware-related for the next.

Sure

**Pull request merged**